Task 3

RamanReut opened the pull request.

RamanReut committed:
Add OrderDetail widget

RamanReut committed:
Merge remote-tracking branch 'origin/task_2' into task_3

RamanReut committed:
Merge branch 'task_2' into task_3

RamanReut committed:
Rework drawer; Write unit tests

RamanReut committed:
Merge branch 'task_2' into task_3

RamanReut committed:
Update dependecies

RamanReut committed:
Fix problem with drawer background

RamanReut committed:
Use data as associative array

RamanReut committed:
Add key prop in DetailItems

RamanReut committed:
Update dependecies

RamanReut committed:
Add unit tests

RamanReut committed:
Add map order status id to status name

RamanReut committed:
Delete unusable component ControlColumnVisibility

RamanReut committed:
Merge branch 'task_3' into task_4

RamanReut committed:
Update dependencies

RamanReut committed:
Merge branch 'task_2' into task_3

RamanReut committed:
Upgrade dependcies

RamanReut committed:
Move visibleColumns into another reducer

RamanReut committed:
Merge branch 'task_3' into task_4

RamanReut committed:
Fix Toolbar label size

RamanReut committed:
Merge branch 'task_3' into task_4

RamanReut committed:
Add Timeline component

RamanReut committed:
Update dependecies

RamanReut committed:
Fix Connector and Step order

RamanReut committed:
Add timeline control by redux

RamanReut committed:
Add disable or enable state to Timeline Step

RamanReut committed:
Update dependencies

RamanReut committed:
Add order's timeline component

RamanReut committed:
Rework expand/collapse animation

RamanReut committed:
Fix text aligns

RamanReut committed:
Improve order's status names

RamanReut committed:
Add dateRangePicker

RamanReut committed:
Fix pick year in DatePicker

RamanReut committed:
Add toolbar for DatePicker

RamanReut committed:
Update dependencies

RamanReut committed:
Update dependecies

RamanReut committed:
Fix wrong behavior on pick month

Don't pick date after pick month

RamanReut committed:
Improve margins

RamanReut committed:
Improve DatePicker color schema

RamanReut committed:
Remove unusable props in DateRangePicker

RamanReut committed:
finish integrate data range filter

RamanReut committed:
Improve breadcrumbs

RamanReut committed:
Remove debug function

dHihich commented on src/features/orders/DetailItemInfo.tsx:
```diff
@@ -0,0 +1,50 @@
+                {quantity.toString()}
+            </DetailShortTextWrapper>
+            <DetailShortTextWrapper title='Discount'>
+                {Math.trunc(discount * 100).toString() + '%'}
```
might be extracted into utils

dHihich commented on src/features/orders/OrderDetail.tsx:
```diff
@@ -0,0 +1,101 @@
+        [dispatch],
+    );
+
+    return (
```
please split this component into smaller local components (located in this file). For instance I'd extract Header and Title components. Now markup looks too 'deep'

dHihich commented on src/features/orders/OrderTabPanel.test.tsx:
```diff
@@ -0,0 +1,102 @@
+    tabClicked = '';
+});
+
+test('should be exist order tab', () => {
```
order tab should exist

dHihich commented on src/features/orders/OrderTabPanel.test.tsx:
```diff
@@ -0,0 +1,102 @@
+    expect(screen.getByRole('tab', { name: 'Order' })).toBeInTheDocument();
+});
+
+test('should be exist customer tab', () => {
```
customer tab should exist

please fix all test names in this file

dHihich commented on src/features/orders/OrderTabPanel.tsx:
```diff
@@ -0,0 +1,113 @@
+                variant='fullWidth'
+            >
+                <Tab 
+                    classes={{ 
```
all tabs have the same classes and disableRipple props., Either define these props on the top and pass it into each tab using spread operator or define your custom local Tab component (located in this file) where these props set by default

dHihich commented on src/features/orders/OrdersTable.tsx:
```diff
@@ -0,0 +1,74 @@
+    const handleRowClick = useCallback((_, { order_id }) => {
+        dispatch(actions.detail.open(order_id))
+    }, [dispatch]);
+    const handleFetch = useCallback(
```
I don't think you should wrap this action dispatch with useCallback. You use it only within useEffect. You should dispatch this fetch action directly in useEffect

dHihich commented on src/features/orders/OrdersTable.tsx:
```diff
@@ -0,0 +1,74 @@
+                components={{
+                    Toolbar: props => (<TableToolbar {...props}></TableToolbar>),
+                }}
+                options={{
```
options might be extracted either into constants file or at least into a variable on the top of this file

dHihich commented on src/features/orders/OrdersTable.tsx:
```diff
@@ -0,0 +1,74 @@
+
+function createExpandableData(data: DataIndexable): Array<types.Row> {
+    const res = new Array<types.Row>();
+    for (let record in data) {
```
better to use `Object.values()`, right?

dHihich commented on src/features/store/store.ts:
```diff
@@ -1,12 +1,12 @@
+import { reducer as orders } from '../orders';
 
 const rootReducer = combineReducers({
     //Add here new reducers
```
remove this comment pls :)

RamanReut committed:
Move discoutToString from DetailItemInfo to utils

RamanReut committed:
Extract Header from OrderDetail

RamanReut committed:
Remove unnecessary useMemo call

RamanReut committed:
Use default tab props

Use default tab props instead of write this props in each Tab

RamanReut committed:
Move handleFetch logic into useEffect

Memoization handleFetch is not required because of
this logic using only in useEffect

RamanReut committed:
Move options into constant TABLE_OPTIONS

RamanReut committed:
Use Object.values instead of for...in

RamanReut committed:
Merge branch 'task_3' into task_4

RamanReut committed:
Remove unnecessary import useMemo

RamanReut committed:
Fix TabPanel width in OrderDetail

RamanReut committed:
Improve transition

RamanReut committed:
Use standart grey color

RamanReut committed:
Update dependencies

RamanReut committed:
Order status use enum

RamanReut committed:
Exctract step generate logic into another function

RamanReut committed:
Remove comment

RamanReut committed:
Fix bad test names

RamanReut committed:
Merge branch 'task_3' into task_4

RamanReut committed:
Fix dependecy problems

RamanReut committed:
Remove unnecessary reducer

RamanReut committed:
Merge branch 'task_4' into range_picker

dHihich commented on src/features/orders/utils.ts:
```diff
@@ -0,0 +1,3 @@
+export function discountToString(discount: number): string {
```
formatPercentages

RamanReut committed:
Merge pull request #9 from Soolamis/breadcrumbs

Improve breadcrumbs

RamanReut committed:
Merge pull request #8 from Soolamis/range_picker

Range picker

RamanReut committed:
Merge pull request #7 from Soolamis/task_4

Task 4
